deepchem/models/torch_models/ferminet.py

```diff
@@ -2,27 +2,268 @@
+    one_up = one_electron[:self.spin[0], :]
+    one_down = one_electron[self.spin[0]:, :]
+
+    for k in range(self.determinant):
+      # spin-up orbitals
+      for i in range(self.spin[0]):
+        for j in range(self.spin[0]):
+          envelope = 0
+          for m in range(self.nucleon_pos.size()[0]):
+            envelope += self.pi_up[k + i + j + m] * torch.exp(
+                -torch.linalg.norm(self.sigma_up[k + i + j + m] *
+                                   one_electron_vector[j][m]))
+          self.psi_up[k][i][j] = (torch.dot(self.w_up[i + k], one_up[j]) +
+                                  self.g_up[i + k]) * envelope
+        # spin-down orbitals
+        for j in range(self.spin[1]):
+          envelope = 0
+          for m in range(self.nucleon_pos.size()[0]):
+            envelope += self.pi_up[k + i + j + m] * torch.exp(
+                -torch.linalg.norm(self.sigma_up[k + i + j + m] *
+                                   one_electron_vector[j][m]))
+          self.psi_down[k][i][j] = (torch.dot(self.w_down[i + k], one_down[j]) +
+                                    self.g_down[i + k]) * envelope
+    psi = torch.sum(torch.det(self.psi_up) * torch.det(self.psi_down))
+    psi_log = 2 * torch.log(torch.abs(psi))
+    return psi_log
+
+  def calculate_potential(self,) -> Any:
+    """Calculates the potential of the molecule system system for to calculate the hamiltonian loss.
+    Returns:
+    --------
+    potential: Any
+      The potential energy of the system.
+    """
+
+    # electron-nuclear potential
+    electron_nuclear_potential = -1 * torch.sum(
+        self.nuclear_charge * (1 / self.one_electron_distance))
+
+    # electron-electron potential
+    electron_electron_potential = torch.sum(
+        torch.tril(1 / self.two_electron_distance, -1))
+
+    # nuclear-nuclear potential
+    charge_shape = self.nuclear_charge.size()
+    nuclear_nuclear_potential = torch.sum(
+        self.nuclear_charge * self.nuclear_charge.reshape(charge_shape[0], 1) *
+        torch.tril(1 / self.inter_atom, -1))
+    potential = electron_nuclear_potential + electron_electron_potential + nuclear_nuclear_potential
+    potential = potential.to(torch.float32)
+    return potential
+
+  def local_energy(self, output: torch.Tensor, input: torch.Tensor) -> Any:
+    """Calculates the hamiltonian of the molecule system.
+    Returns:
+    --------
+    hamiltonian: Any
+      The hamiltonian of the system.
+    """
+    jacobian = torch.autograd.grad([output], [input], retain_graph=True)
+    hessian = torch.autograd.functional.hessian(self.forward,
+                                                input,
+                                                retain_graph=True)
+
+    potential = self.calculate_potential()
+    return potential - 0.5 * (torch.sum(torch.diag(hessian, 0)) +
+                              torch.sum(torch.pow(jacobian, 2)))
+
+
+class FerminetModel(TorchModel):
```

Comment:
As Bharath suggested, you can override whatver TorchModel method is preventing you from running ElectronSampler between iterations.